mloop/learners.py

```diff
@@ -1018,15 +1058,28 @@ def _rand2(self, index):
 
     def random_index_sample(self, index, num_picks):
         '''
-        Randomly select a num_picks of indexes, without index.
+        Randomly select a num_picks of indexes, without index, from 
+        a subset of all cases given by self.elite, where 1 selects from
+        all and 0 gives the num_picks best
 
         Args:
             index(int): The index that is not included
             num_picks(int): The number of picks.
         '''
-        rand_indexes = list(range(self.num_population_members))
-        rand_indexes.remove(index)
-        return random.sample(rand_indexes, num_picks)
+        
+        # Begin by removing index from list of costs and indices
+        rand_costs = np.delete(self.population_costs, index) # np.delete returns a copy
+        rand_indexes = np.delete(np.arange(self.num_population_members), index)
+
+        locations = np.argsort(rand_costs)
+        rand_indexes = rand_indexes[locations] # indices sorted by increasing cost
+        
+        # now remove a fraction given by elite
+        num = math.ceil(self.num_population_members*self.elite)
+        if num < num_picks: num = num_picks
+        
+        
+        return random.sample(rand_indexes[:num], num_picks)
```

Comment:
```suggestion
        return random.sample(list(rand_indexes[:num]), num_picks)
```

See [this comment](https://github.com/michaelhush/M-LOOP/pull/107#issuecomment-902277704) for reasoning behind this requested change.